resources/assets/coffee/react/mp-history/content.coffee

```diff
@@ -0,0 +1,88 @@
+###
+# Copyright 2016 ppy Pty. Ltd.
+#
+# This file is part of osu!web. osu!web is distributed with the hope of
+# attracting more community contributions to the core ecosystem of osu!.
+#
+# osu!web is free software: you can redistribute it and/or modify
+# it under the terms of the Affero GNU General Public License version 3
+# as published by the Free Software Foundation.
+#
+# osu!web is distributed WITHOUT ANY WARRANTY; without even the implied
+# warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+###
+{div, a} = React.DOM
+el = React.createElement
+
+class MPHistory.Content extends React.Component
+  teamScores: (eventIndex) ->
+    @scoresCache ?= {}
+
+    if !@scoresCache[eventIndex]?
+      left = 0
+      right = 0
+
+      for score in @props.events[eventIndex].game.data.scores.data
+        if score.team == 0
+          left += score.score
+        else
+          right += score.score
+
+      @scoresCache[eventIndex] =
+        left: left
+        right: right
+
+    return @scoresCache[eventIndex]
+
+  showMore: (e) ->
+    e.preventDefault()
+    $.publish 'events:show-more'
+
+  render: ->
+    if _.isEmpty @props.events
+      div className: 'osu-layout__row osu-layout__row--page-mp-history',
+        Lang.get 'multiplayer.match.loading-events'
+    else
+      gameIds = _(@props.events)
+        .map (m, i) ->
+          return if !m.game?
+          i
+        .filter (m) -> m?
+        .value()
+
+        gameIds = [-1, gameIds...]
+
+        # grabbing events that are past the last game
+        lastEvents = @props.events[_.last(gameIds) + 1..@props.events.length]
+
+      div className: 'osu-layout__row osu-layout__row--page-mp-history',
+        if @props.remainingEventsCount > 0
+          div className: 'mp-history-content__show-more-box',
+            a
+              className: 'mp-history-content__show-more'
+              onClick: @showMore
+              Lang.get 'multiplayer.match.more-events', count: @props.remainingEventsCount
+
+        for id, i in gameIds
```

Comment:
well, the events need to be wrapped in a separate box like in [here](http://puu.sh/plb6g.png), so I can't render them like that in a separate loop